crates/env/src/backend.rs

```diff
@@ -302,7 +302,7 @@ pub trait TypedEnvBackend: EnvBackend {
     /// # Note
     ///
     /// For more details visit: [`ink_env::terminate_contract`]
-    fn terminate_contract<T>(&mut self, beneficiary: T::AccountId) -> !
+    fn terminate_contract<T>(&mut self, beneficiary: T::AccountId) -> Result<()>
```

Comment:
Not sure about the return type here. I guess the real thing would indeed never return.

Reply:
I guess you changed this API to make it possible to deliver an off-chain implementation of it. However, as you already noticed our off-chain environment is kind of limited and the inability to do proper contract terminations is one. Instead of adjusting useful interfaces to an non useful off-chain implementation we should instead strife for improving the off-chain environment so that it is able to handle those things if possible.

Reply:
This means that for example contracts we cannot really off-chain test them properly if they make use of such behavior.

Reply:
You example contract is fine to cover both APIs. However, I think for `transferred_balance` we could also create a simple alternative example contract similar to what I did in the release notes of v3.0.0-rc1.